tests/proxy.spec.coffee

```diff
@@ -0,0 +1,95 @@
+_ = require('lodash')
+m = require('mochainon')
+globalTunnel = require('global-tunnel-ng')
+EventEmitter = require('events')
+child_process = require('child_process')
+path = require('path')
+
+IS_BROWSER = window?
+
+{ token, getCustomRequest } = require('./setup')()
+request = getCustomRequest(null, false)
+
+startProxy = (secure) ->
+	events = new EventEmitter()
+	scriptPath = path.join(__dirname, 'proxy', 'start-proxy.js')
+	proxy = child_process.spawn('node', [ scriptPath, if secure then '1' else '' ])
+
+	proxy.stdout.on 'data', (line) ->
+		line = line.toString().trim()
+		sep = line.indexOf(':')
+		if sep >= 0
+			event = line.substring(0, sep)
+			arg = line.substring(sep + 1)
+		else
+			event = line
+			arg = null
+		events.emit(event, arg)
+
+	proxy.on 'exit', ->
+		events.emit('stop')
+
+	return {
+		on: events.on.bind(events)
+		stop: ->
+			proxy.kill()
+	}
+
+describe 'Proxy support', ->
+	return if IS_BROWSER
+	@timeout(10000)
+
+	describe 'HTTP proxy', ->
+		proxy = null
+		proxyReached = false
```

Comment:
This should be set in a `beforeEach`. If we add another test in this block sometime, `proxyReached` will already be true.

Also, to nitpick, you don't need to pull these two variables into the outer scope anyway, I think the idiomatic mocha route is to use `@proxy = ` inside the before/after. I don't really mind though.